packages/core/components/FileMetadataSearchBar/index.tsx

```diff
@@ -43,6 +42,28 @@ export function extractDateFromDateString(dateString?: string): Date | undefined
     return date;
 }
 
+function extractDatesFromRangeOperatorFilterString(
+    filterString: string
+): { startDate: Date; endDate: Date } | null {
+    // Regex with capture groups for identifying ISO datestrings in the RANGE() filter operator
+    // e.g. RANGE(2022-01-01T00:00:00.000Z,2022-01-31T00:00:00.000Z)
+    // Captures "2022-01-01T00:00:00.000Z" and "2022-01-31T00:00:00.000Z"
+    const RANGE_OPERATOR_REGEX = /RANGE\(([\d\-:TZ.]+),([\d\-:TZ.]+)\)/g;
+    const exec = RANGE_OPERATOR_REGEX.exec(filterString);
+    if (exec && exec.length === 3) {
+        // Length of 3 because we use two capture groups
+        const startDate = new Date(exec[1]);
+        // The RANGE() filter uses an exclusive upper bound.
+        // However, we want to present dates in the UI as if the upper bound was inclusive.
+        // To handle that, we'll subtract a day from the upper bound used by the filter, then present the result
```

Comment:
wait, is upper bound the most recent?  in dates, upper usually means more milliseconds.  Subtracting a day would be the wrong thing.

Reply:
So if I have a filter like `RANGE(day1, day2)`, "day1" is the lower bound, and "day2" is the upper bound. The upper bound in this query is exclusive, following the same logic as Python's built-in `range()` function (e.g. `list(range(0, 10))` generates a list of numbers starting with 0 and ending with 9).
However, in the File Explorer App, it would be unintuitive for users to select their exclusive upper bound via the UI we have. I wanted to allow users to pick an _inclusive_ upper bound in the UI, which means if we want to use this RANGE() filter in generating queries we have to do a little behind-the-scenes magic to make that work. So when they pick their upper bound, the code does a +1 to make the filter work, and when rendering the UI based off of an existing filter the code does a -1 to make the UI display what the user would expect